master/internal/core.go

```diff
@@ -168,6 +169,14 @@ func (m *Master) startServers() error {
 	runServer := func(server *http.Server) {
 		errs <- errors.Wrap(m.echo.StartServer(server), servers[server]+" failed")
 	}
+	go func() {
+		if err := grpc.RegisterHTTPProxy(m.echo, m.config.GRPCPort); err != nil {
```

Comment:
question: can we explain why we are doing something slightly different here than than the other ones?

As in, why don't we do `errs <- grpc.RegisterHTTPProxy(m.echo, m.config.GRPCPort)`?

Reply:
RegisterHTTPProxy does not block; it just adds the endpoints to the echo server. In other words it does not start its own server. All the other calls start a server and block until there is a fatal error.